lib/spack/spack/build_environment.py

```diff
@@ -411,7 +409,7 @@ def _order(specs):
         x.prefix for x in build_link_deps)
 
     # Add dependencies to CMAKE_PREFIX_PATH
-    env.set_path('CMAKE_PREFIX_PATH', build_link_prefixes)
+    env.set_path('CMAKE_PREFIX_PATH', get_cmake_prefix_path(pkg))
```

Comment:
Shouldn't this just be

```suggestion
    env.set_path('CMAKE_PREFIX_PATH', filter_system_paths(x.prefix  for x in build_link_deps))
```

Then we don't need to do all the recomputing of `build_link_deps` in the method below.

Reply:
Sure, but I want to avoid that in the future the env variable and the cli version diverge again

Reply:
I don't see any performance issues here, e.g. with rocfft concretizer in an environment #23859:

Using get_cmake_prefix_path:

```
harmen-desktop tmp.14BtIQnfi5 $ hyperfine 'spack -e . build-env rocfft'
Benchmark #1: spack -e . build-env rocfft
  Time (mean ± σ):      1.652 s ±  0.015 s    [User: 1.521 s, System: 0.130 s]
  Range (min … max):    1.630 s …  1.676 s    10 runs
```

Using filter_system_paths(...):
```
harmen-desktop tmp.14BtIQnfi5 $ hyperfine 'spack -e . build-env rocfft'
Benchmark #1: spack -e . build-env rocfft
  Time (mean ± σ):      1.656 s ±  0.014 s    [User: 1.523 s, System: 0.132 s]
  Range (min … max):    1.629 s …  1.673 s    10 runs
```

Reply:
Ok, but we should not be constructing the same list the same way in two different places. That's how we got into this mess in the first place.

Can you create a helper function `get_build_link_deps` that can be called in both places?